### PR TITLE
Character Creation Ease

### PR DESCRIPTION
**Pull request by KPrasch**
Implements some of the proposed ideas here: <https://github.com/nucypher/nucypher/pull/1338#issuecomment-537176316>

```python
#
# Setup
#

PROVIDER_URI = '~/.ethereum/goerli/geth.ipc'
KEYRING_ROOT = "~/.local/share/nucypher/keyring"
ACCOUNT = ''
PASSWORD = ''

SEEDNODES = [
   ...
]

#
# Keyring
#

try:
    # Create a new Keyring
    keyring = NucypherKeyring.generate(keyring_root=KEYRING_ROOT, password=PASSWORD, checksum_address=ACCOUNT)
except ExistingKeyringError:
    # Use an existing Keyring
    keyring = NucypherKeyring(keyring_root=KEYRING_ROOT, account=ACCOUNT)
keyring.unlock(password=PASSWORD)


#
# Alice
#

from nucypher.characters.lawful import Alice

alice = Alice(keyring=keyring, known_nodes=SEEDNODES, provider_uri=PROVIDER_URI)
alice.start_learning_loop(now=True)

# alice.grant()
```

### CODE REVIEW

File: nucypher/characters/base.py [hunk 1]
```diff
@@ -138,13 +156,12 @@ def __init__(self,
         #
         # Self-Character
         #
-        if is_me is True:
 
+        if is_me is True:
```
Comment (michwill) [inline nucypher/characters/base.py]: Do we need `is_me is True`, not just `is_me` here?

Comment (KPrasch) [reply nucypher/characters/base.py]: Yeah - I think just `is_me` will suffice.

File: nucypher/blockchain/eth/interfaces.py [hunk 2]
```diff
@@ -745,15 +756,15 @@ def get_interface(cls, provider_uri: str = None) -> Interfaces:
...
+                raise cls.InterfaceNotInitialized(f"There is no connection for {provider_uri}. "
+                                                  f"Call .initialize_connection, then try again.")
 
         # Try to use the most recently created interface by default.
         else:
             try:
                 cached_interface = list(cls._interfaces.values())[-1]
             except IndexError:
-                raise cls.FactoryError(f"There is no existing blockchain connection.")
+                raise cls.NoRegisteredInterfaces(f"There is no existing blockchain connection.")
```
[the remark this answers is not in the document]
Comment (KPrasch) [reply nucypher/blockchain/eth/interfaces.py]: :heavy_check_mark:

File: nucypher/blockchain/eth/interfaces.py [hunk 3]
```diff
@@ -745,15 +756,15 @@ def get_interface(cls, provider_uri: str = None) -> Interfaces:
             try:
                 cached_interface = cls._interfaces[provider_uri]
             except KeyError:
-                raise cls.FactoryError(f"There is no connection for {provider_uri}. "
-                                       f"Call .initialize_connection, then try again.")
+                raise cls.InterfaceNotInitialized(f"There is no connection for {provider_uri}. "
+                                                  f"Call .initialize_connection, then try again.")
```
[the remark this answers is not in the document]
Comment (KPrasch) [reply nucypher/blockchain/eth/interfaces.py]: :heavy_check_mark:

File: nucypher/blockchain/eth/interfaces.py [hunk 4]
```diff
@@ -727,7 +737,8 @@ def initialize_interface(cls,
                              ) -> None:
 
         if provider_uri in cls._interfaces:
-            raise cls.FactoryError(f"A connection already exists for {provider_uri}.  Use .get_interface instead.")
+            raise cls.InterfaceAlreadyInitialized(f"A connection already exists for {provider_uri}.  "
```
[the remark this answers is not in the document]
Comment (KPrasch) [reply nucypher/blockchain/eth/interfaces.py]: :heavy_check_mark:

File: nucypher/blockchain/eth/interfaces.py [hunk 5]
```diff
@@ -712,7 +721,8 @@ def register_interface(cls,
 
         provider_uri = interface.provider_uri
         if provider_uri in cls._interfaces:
-            raise cls.FactoryError(f"A connection already exists for {provider_uri}.  Use .get_interface instead.")
+            raise cls.InterfaceAlreadyInitialized(f"A connection already exists for {provider_uri}. "
+                                                  f" Use .get_interface instead.")
```
[the remark this answers is not in the document]
Comment (KPrasch) [reply nucypher/blockchain/eth/interfaces.py]: :heavy_check_mark:

File: nucypher/characters/base.py (hunk 1 above)
Comment (KPrasch) [reply nucypher/characters/base.py]: :heavy_check_mark:

File: nucypher/characters/lawful.py [hunk 6]
```diff
@@ -1049,7 +1051,7 @@ def __attempt(attempt=1, interval=10) -> Ursula:
     @classmethod
     def from_seed_and_stake_info(cls,
                                  seed_uri: str,
-                                 federated_only: bool,
+                                 federated_only: bool = False,
```
Comment (tuxxy) [inline nucypher/characters/lawful.py]: :cowboy_hat_face:

File: nucypher/config/keyring.py [hunk 7]
```diff
@@ -510,8 +515,8 @@ def derive_crypto_power(self, power_class: ClassVar) -> Union[KeyPairBasedPower,
     def generate(cls,
                  checksum_address: str,
                  password: str,
-                 encrypting: bool,
-                 rest: bool,
+                 encrypting: bool = True,
+                 rest: bool = False,
                  host: str = None,
                  curve: EllipticCurve = None,
                  keyring_root: str = None,
```
Comment (tuxxy) [inline nucypher/config/keyring.py]: This is a bit awkward. Seems a bit odd to take an assumption on _what_ the keypair will do (e.g. encrypting), but not the curve to do it with.

**Pull request merged**